### PR TITLE
(HC-97) Remove library prefixing.  Add support to build libcpp-hocon …

### PR DESCRIPTION
**Pull request by jmslagle**
…shared

### CODE REVIEW

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -75,20 +75,23 @@ set(PROJECT_SOURCES
 ## Without the intermediate target, unexported symbols can't be tested.
 add_library(libprojectsrc OBJECT ${PROJECT_SOURCES})
 set_target_properties(libprojectsrc PROPERTIES POSITION_INDEPENDENT_CODE true)
-
-add_library(lib${PROJECT_NAME} $<TARGET_OBJECTS:libprojectsrc>)
-set_target_properties(lib${PROJECT_NAME} PROPERTIES VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
-target_link_libraries(lib${PROJECT_NAME}
+if(CPP_HOCON_SHARED)
```
Comment (MikaelSmith) [inline lib/CMakeLists.txt]: This is actually unnecessary, you can use https://cmake.org/cmake/help/v3.0/variable/BUILD_SHARED_LIBS.html

File: lib/src/config_document_parser.cc [hunk 2]
```diff
@@ -33,8 +33,7 @@ namespace hocon { namespace config_document_parser {
 
     /** Parse context */
     parse_context::parse_context(config_syntax flavor, shared_origin origin, token_iterator tokens) :
-        _line_number(1), _tokens(move(tokens)), _flavor(flavor), _base_origin(move(origin)),
-        _equals_count(0) { }
```
Comment (MikaelSmith) [inline lib/src/config_document_parser.cc]: Why is this removed? It should still be initialized (C++ doesn't default initialize ints to 0).

**Comment by MikaelSmith** [comment]
Thanks for working on this fix!

File: lib/CMakeLists.txt (hunk 1 above)
Comment (jmslagle) [inline lib/CMakeLists.txt]: Aha!  I can nuke that - I just copied what Leatherman is going :)

File: lib/src/config_document_parser.cc (hunk 2 above)
Comment (jmslagle) [inline lib/src/config_document_parser.cc]: Erm?  I actually have no clue where those went.  I'm guessing they accidently got nuked as I went through and fixed all the LIBCPP atuff.  I'll fix.

**Comment by jmslagle** [comment]
I've made the changes.  I suspect it's just an oddity with when I pulled my initial branch for HC-95 on the variable thing.  The CMake trickery I didn't know and am updating my port.

**Comment by paravoid** [comment]
I basically did the very same, unaware of this :) This is the fallout from puppetlabs/leatherman#213 which in turn is pretty much required for packaging them for Debian.

**Comment by MikaelSmith** [comment]
@puppetlabs/night-s-watch do you have some bandwidth to deal with any downstream effects of this? I think everything will still find the library, but not 100% sure.

**Comment by GabrielNagy** [comment]
Hi @MikaelSmith,

I started an agent build with this change and https://github.com/puppetlabs/leatherman/pull/213, everything compiles fine (facter, leatherman, libwhereami) until cpp-pcp-client which gets installed to `liblibcpp-pcp-client.so` and pxp-agent cannot find it.

I assume something similar must be also done on cpp-pcp-client?

**Comment by GabrielNagy** [comment]
@smortex unfortunately this change greatly impacts how we build our puppet-agent packages.

Since we're looking to remove cpp projects from our pipeline (as we're introducing [facter-ng](https://github.com/puppetlabs/facter-ng/)) I'm not sure we want to make these changes (impacted projects are cpp-hocon, cpp-pcp-client, leatherman, libwhereami, facter, and still I ran into some issues with libfacter.so not finding libraries).

**Comment by GabrielNagy** [comment]
@jmslagle thanks for the understanding, I'll go ahead and close this then.
